include/ginkgo/core/base/executor.hpp

```diff
@@ -1532,7 +1541,8 @@ class CudaExecutor : public detail::ExecutorBase<CudaExecutor>,
     static std::shared_ptr<CudaExecutor> create(
         int device_id, std::shared_ptr<Executor> master,
         bool device_reset = false,
-        allocation_mode alloc_mode = default_cuda_alloc_mode);
+        allocation_mode alloc_mode = default_cuda_alloc_mode,
+        CUstream_st* stream = nullptr);
```

Comment:
Perhaps just as a reminder, I think it would be good, to have only constructors with either device_reset or stream parameter.

Reply:
maybe it would make sense to handle this in another PR adding custom allocator support (for AMReX's memory pool)

Reply:
The issue is not with the allocation mode, but with the device reset. So I think it fits in here. I think at the top of the PR there is our old discussion, and this was just meant as a reminder for that.

Reply:
Yes, I only want to keep the number of interface changes small, since this will likely be another longer discussion :smile:

Reply:
It still anticipates a design decision we haven't made yet. We should probably discuss that tomorrow :)

Reply:
I think I'd prefer sticking with this for now, and changing the new constructor in a subsequent PR